src/containers/forms/PlanForm/tests/helpers.test.ts

```diff
@@ -0,0 +1,144 @@
+      expectedActivity.bednetDistribution
+    );
+    expect(extractActivityForForm(planActivityWithEmptyfields.bednetDistribution)).toEqual(
+      expectedActivityEmptyField.bednetDistribution
+    );
+  });
+
+  it('check extractActivityForForm returns the correct value for bloodScreening', () => {
+    expect(extractActivityForForm(planActivities.bloodScreening)).toEqual(
+      expectedActivity.bloodScreening
+    );
+    expect(extractActivityForForm(planActivityWithEmptyfields.bloodScreening)).toEqual(
+      expectedActivityEmptyField.bloodScreening
+    );
+  });
+
+  it('check extractActivityForForm returns the correct value for caseConfirmation', () => {
+    expect(extractActivityForForm(planActivities.caseConfirmation)).toEqual(
+      expectedActivity.caseConfirmation
+    );
+    expect(extractActivityForForm(planActivityWithEmptyfields.caseConfirmation)).toEqual(
+      expectedActivityEmptyField.caseConfirmation
+    );
+  });
+
+  it('check extractActivityForForm returns the correct value for larvalDipping', () => {
+    expect(extractActivityForForm(planActivities.larvalDipping)).toEqual(
+      expectedActivity.larvalDipping
+    );
+    expect(extractActivityForForm(planActivityWithEmptyfields.larvalDipping)).toEqual(
+      expectedActivityEmptyField.larvalDipping
+    );
+  });
+
+  it('check extractActivityForForm returns the correct value for familyRegistration', () => {
+    expect(extractActivityForForm(planActivities.familyRegistration)).toEqual(
+      expectedActivity.familyRegistration
+    );
+  });
+
+  it('check extractActivityForForm returns the correct value for mosquitoCollection', () => {
+    expect(extractActivityForForm(planActivities.mosquitoCollection)).toEqual(
+      expectedActivity.mosquitoCollection
+    );
+    expect(extractActivityForForm(planActivityWithEmptyfields.mosquitoCollection)).toEqual(
+      expectedActivityEmptyField.mosquitoCollection
+    );
+  });
+
+  it('check getFormActivities returns the correct value', () => {
+    expect(JSON.stringify(getFormActivities(planActivities))).toEqual(
+      JSON.stringify(extractedActivitiesFromForms)
+    );
+  });
+
+  it('check doesFieldHaveErrors returns the correct value', () => {
+    let errors = [
+      {
+        id: 'Required',
+      },
+    ];
+    let field = 'id';
+    let index = 0;
+    expect(doesFieldHaveErrors(field, index, errors)).toBe(true);
+    field = '';
+    index = NaN;
+    errors = [];
+    expect(doesFieldHaveErrors(field, index, errors)).toBe(false);
+  });
+
+  it('check extractActivitiesFromPlanForm returns the correct value', () => {
+    MockDate.set('1/30/2000', 0);
+    expect(extractActivitiesFromPlanForm(activities)).toEqual(
+      expectedExtractActivityFromPlanformResult
+    );
+    MockDate.reset();
+  });
+
+  it('check getNameTitle returns the correct value when Focus Investigation(FI) is selected', () => {
+    expect(getNameTitle(event, values)).toEqual(['A1--2019-08-09', 'A1  2019-08-09']);
```

Comment:
Two things:

1. The generated value is missing the jurisdiction name.
2. Where does the date `2019-08-09` come from?

Reply:
When the getNameTitle function is called when a location is not selected; then a jurisdiction name will be missing as you see here. But on the following line, you see that the function is called when a location is selected.

The date comes from here: https://github.com/onaio/reveal-frontend/blob/master/src/containers/forms/PlanForm/helpers.ts#L324 and/or here: https://github.com/onaio/reveal-frontend/blob/master/src/containers/forms/PlanForm/helpers.ts#L315. As you can see, it gets the current date

Reply:
> When the getNameTitle function is called when a location is not selected; then a jurisdiction name will be missing as you see here. But on the following line, you see that the function is called when a location is selected.

Thanks.

> The date comes from here: https://github.com/onaio/reveal-frontend/blob/master/src/containers/forms/PlanForm/helpers.ts#L324 and/or here: https://github.com/onaio/reveal-frontend/blob/master/src/containers/forms/PlanForm/helpers.ts#L315. As you can see, it gets the current date

Thanks.  However the value there is not the current date.  Where does **that** value come from?

Reply:
It comes from the fixtures